src/analyzer/protocol/ssl/spicy/SSL.spicy

```diff
@@ -0,0 +1,2058 @@
+    TLS_DHE_PSK_WITH_CAMELLIA_256_CBC_SHA384 = 0xC097,
+    TLS_RSA_PSK_WITH_CAMELLIA_128_CBC_SHA256 = 0xC098,
+    TLS_RSA_PSK_WITH_CAMELLIA_256_CBC_SHA384 = 0xC099,
+    TLS_ECDHE_PSK_WITH_CAMELLIA_128_CBC_SHA256 = 0xC09A,
+    TLS_ECDHE_PSK_WITH_CAMELLIA_256_CBC_SHA384 = 0xC09B,
+    # RFC 6655
+    TLS_RSA_WITH_AES_128_CCM = 0xC09C,
+    TLS_RSA_WITH_AES_256_CCM = 0xC09D,
+    TLS_DHE_RSA_WITH_AES_128_CCM = 0xC09E,
+    TLS_DHE_RSA_WITH_AES_256_CCM = 0xC09F,
+    TLS_RSA_WITH_AES_128_CCM_8 = 0xC0A0,
+    TLS_RSA_WITH_AES_256_CCM_8 = 0xC0A1,
+    TLS_DHE_RSA_WITH_AES_128_CCM_8 = 0xC0A2,
+    TLS_DHE_RSA_WITH_AES_256_CCM_8 = 0xC0A3,
+    TLS_PSK_WITH_AES_128_CCM = 0xC0A4,
+    TLS_PSK_WITH_AES_256_CCM = 0xC0A5,
+    TLS_DHE_PSK_WITH_AES_128_CCM = 0xC0A6,
+    TLS_DHE_PSK_WITH_AES_256_CCM = 0xC0A7,
+    TLS_PSK_WITH_AES_128_CCM_8 = 0xC0A8,
+    TLS_PSK_WITH_AES_256_CCM_8 = 0xC0A9,
+    TLS_PSK_DHE_WITH_AES_128_CCM_8 = 0xC0AA,
+    TLS_PSK_DHE_WITH_AES_256_CCM_8 = 0xC0AB,
+    TLS_ECDHE_ECDSA_WITH_AES_128_CCM = 0xC0AC,
+    TLS_ECDHE_ECDSA_WITH_AES_256_CCM = 0xC0AD,
+    TLS_ECDHE_ECDSA_WITH_AES_128_CCM_8 = 0xC0AE,
+    TLS_ECDHE_ECDSA_WITH_AES_256_CCM_8 = 0xC0AF,
+    # draft-agl-tls-chacha20poly1305-02
+    TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256_OLD = 0xCC13,
+    TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256_OLD = 0xCC14,
+    TLS_DHE_RSA_WITH_CHACHA20_POLY1305_SHA256_OLD = 0xCC15,
+    # RFC 7905
+    TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256 = 0xCCA8,
+    TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256 = 0xCCA9,
+    TLS_DHE_RSA_WITH_CHACHA20_POLY1305_SHA256 = 0xCCAA,
+    TLS_PSK_WITH_CHACHA20_POLY1305_SHA256 = 0xCCAB,
+    TLS_ECDHE_PSK_WITH_CHACHA20_POLY1305_SHA256 = 0xCCAC,
+    TLS_DHE_PSK_WITH_CHACHA20_POLY1305_SHA256 = 0xCCAD,
+    TLS_RSA_PSK_WITH_CHACHA20_POLY1305_SHA256 = 0xCCAE,
+    # draft-ietf-tls-ecdhe-psk-aead-05 & RFC8442
+    TLS_ECDHE_PSK_WITH_AES_128_GCM_SHA256 = 0xD001,
+    TLS_ECDHE_PSK_WITH_AES_256_GCM_SHA384 = 0xD002,
+    TLS_ECDHE_PSK_WITH_AES_128_CCM_8_SHA256 = 0xD003,
+    TLS_ECDHE_PSK_WITH_AES_128_CCM_SHA256_OLD = 0xD004,
+    TLS_ECDHE_PSK_WITH_AES_128_CCM_SHA256 = 0xD005,
+};
+
+type Share = unit {
+    # is the direction not the normal way round?
+    var flipped: bool;
+    var flip_already_alerted: bool;
+    # version as seen in server_hello (for signature and hash-alg choice)
+    var chosen_version_sh_outer: uint16;
+    # final negotiated version - can e.g. be used to distinguished tls 1.3
+    var negotiated_version: uint16;
+    # set to true if chosen version is identified as a tls 1.3 version
+    var tls_13: bool;
+    var chosen_cipher: uint16;
+    var ccs_seen: uint8;
+    var invalid_dtls_version_count: uint32;
+    #  var skipping: bool;
+    var client_encrypted: bool;
+    var server_encrypted: bool;
+    var both_sides_encrypted_first_time: bool;
+    var established: bool;
+    var client_certificate_depth: uint32;
+    var server_certificate_depth: uint32;
+
+    on %init {
```

Comment:
Total flyby. `%init` doesn't actually do anything when used as %context. Seems everything is initialized to default values anyhow?

https://github.com/zeek/spicy/issues/1502 (think this can be closed).

I'm also wondering if we should come up with a consistent name for structs used for `%context`, I've used `Ctx` for LDAP. In QUIC it's `ConnectionIDInfo` for historical reasons, I'd use Ctx today. Ctx or Context seems the least amount of cognitive overhead.

Reply:
Thank you - I totally was not aware of this.